tests/helpers/validators.py

```diff
@@ -83,3 +89,29 @@ def test_ip_or_domain(self):
 
         with self.assertRaises(argparse.ArgumentTypeError):
             ArgumentType.ip_or_domain("Definitely not a host")
+
+    def test_comma_separated_integers_or_file(self):
+
+        with self.assertRaises(argparse.ArgumentTypeError):
+            ArgumentType.comma_separated_integers_or_file("/dev/null/fail")
+
+        with self.assertRaises(argparse.ArgumentTypeError):
+            ArgumentType.comma_separated_integers_or_file("not,a,number")
+
+        with self.assertRaises(argparse.ArgumentTypeError):
+            ArgumentType.comma_separated_integers_or_file("1, 2, 3")
+
+        old = sys.stdin
+        sys.stdin = StringIO("1\n2\n")
+        self.assertEqual(
+            ArgumentType.comma_separated_integers_or_file("-"),
+            [1, 2]
+        )
+        sys.stdin = old
+
+        with open("/tmp/__test_file__", "w") as f:
+            f.write("1\n2\n3\n")
+        self.assertEqual(
+            ArgumentType.comma_separated_integers_or_file("/tmp/__test_file__"),
+            [1, 2, 3]
+        )
```

Comment:
maybe we should remove the file after we finish? it's still tmp but it would be weird if someone sees that file with content, without knowing where it came from :)